catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FindChildrenStreamEndPlugin.java

```diff
@@ -0,0 +1,168 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.plugins;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.codice.alliance.video.stream.mpegts.Context;
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.impl.SortByImpl;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * Find the children of the parent metacard and process the children in batches. The children will be
+ * processed in the order in which they were originally created.
+ */
+public class FindChildrenStreamEndPlugin implements StreamEndPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FindChildrenStreamEndPlugin.class);
+
+    private static final int BATCH_SIZE = 50;
+
+    private final Factory factory;
+
+    public FindChildrenStreamEndPlugin(Factory factory) {
+        this.factory = factory;
+    }
+
+    public Factory getFactory() {
+        return factory;
+    }
+
+    @Override
+    public void streamEnded(Context context) {
+
+        synchronized (context.isParentDirty()) {
+
+            if (!context.isParentDirty()
+                    .get()) {
+                return;
+            }
+
+            context.getParentMetacard()
+                    .ifPresent(parentMetacard -> handleParentMetacard(context, parentMetacard));
+
+            context.isParentDirty()
+                    .set(false);
+        }
+    }
+
+    @Override
+    public void accept(Visitor visitor) {
+        visitor.visit(this);
+    }
+
+    private void handleParentMetacard(Context context, Metacard parentMetacard) {
+
+        Handler handler = factory.build();
+
+        try {
+
+            final FilterBuilder builder = new GeotoolsFilterBuilder();
```

Comment:
future enhancement: I think this is something that we should be injecting so we don't rely on the geotools implementation.